packages/truffle-decoder/lib/decode/memory.ts

```diff
@@ -7,21 +7,21 @@ import decodeValue from "./value";
 import { MemoryPointer, DataPointer } from "../types/pointer";
 import { MemoryMemberAllocation } from "../types/allocation";
 import { EvmInfo } from "../types/evm";
-import range from "lodash.range";
+import { DecoderRequest } from "../types/request";
 
-export default async function decodeMemory(definition: DecodeUtils.AstDefinition, pointer: MemoryPointer, info: EvmInfo): Promise <any> {
+export default function* decodeMemory(definition: DecodeUtils.AstDefinition, pointer: MemoryPointer, info: EvmInfo): IterableIterator<any | DecoderRequest> {
```

Comment:
Did you look if there's been any prior effort towards doubly-parameterized IterableIterators? Maybe someone's found a workaround or opened an improvement proposal?

Reply:
Interesting question.  I'll see what I can find (though I'm not expecting much).  Maybe I could even write a proposal.

Btw, note that the line you highlighted isn't actually an example of this problem, although it will be once we have the new output format.  Writing `any | DecoderRequest` isn't actually necesary; I just included `DecoderRequest` for clarity.  The ugly part isn't having to include `DecoderRequest` (because including that makes sense, that *is* a possible value that could be yielded), it's having to include `Uint8Array`.

Reply:
Oh hey, turns out TypeScript is on GitHub, guess I can try filing an issue at least...

Reply:
Oh, turns out there is in fact a TypeScript PR for this! (along with some other things) https://github.com/Microsoft/TypeScript/pull/30790